sim/dex-conditions.ts

```diff
@@ -72,7 +72,9 @@ export interface EventMethods {
 		this: Battle, target: Pokemon, source: Pokemon, source2: Effect, move: ActiveMove
 	) => Pokemon | void;
 	onResidual?: (this: Battle, target: Pokemon, source: Pokemon, effect: Effect) => void;
-	onSetAbility?: (this: Battle, ability: string, target: Pokemon, source: Pokemon, effect: Effect) => boolean | void;
+	onSetAbility?: (
+		this: Battle, ability: string, target: Pokemon, source: Pokemon, effect: Effect
+	) => boolean | null | void;
```

Comment:
I was wondering why there was both boolean and null here and noticed that there is some Protective Pads code in mods that still tries to use this. Is that left over from your changes in PR #8207? It looks as if the code there won't get called because the check for a contact move would already have failed.